### DISCUSSION TITLE
What if I have a source folder external to my project folder?

### DISCUSSION THREAD

**Discussion by ubaldot** [clangd/clangd#1929] [Q&A] [answered]
Say that I have this folder structure: 

```
├── components
├── stm32_platform
           └── Hello_World
                       └──  compile_commands.json
```

and assume that I have a `compile_commands.json` in `./stm32_platfom/hello_world`, which is my project root, how to make `clangd` to work well for `.c/.h `files located in `./components`?

**Comment by HighCommander4** [comment, marked as answer]
Clangd has a `--compile-commands-dir` command-line argument, so that if you run clangd with `--compile-commands-dir=/path/to/stm32_platform/Hello_World`, it will use the `compile_commands.json` in that directory for all opened files, including the ones in `/components`.
